x/cronos/keeper/keeper.go

```diff
@@ -215,3 +211,47 @@ func (k Keeper) IbcTransferCoins(ctx sdk.Context, from, destination string, coin
 	}()
 	return nil
 }
+
+// GetExternalContractByDenom find the corresponding contract for the denom,
+// external contract is taken in preference to auto-deployed one
+func (k Keeper) GetExternalContractByDenom(ctx sdk.Context, denom string) (common.Address, bool) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.DenomToExternalContractKey(denom))
+	if len(bz) == 0 {
+		return common.Address{}, false
+	}
+
+	return common.BytesToAddress(bz), true
+}
+
+// GetAutoContractByDenom find the corresponding contract for the denom,
+// external contract is taken in preference to auto-deployed one
+func (k Keeper) GetAutoContractByDenom(ctx sdk.Context, denom string) (common.Address, bool) {
```

Comment:
I think managing two mapping for autocontract and externalcontract is confusing.

One is enough. The app does not need to manage auto <-> external

Reply:
The reason for the two mappings is when we have auto contract and external contract at the same time for a denom, we still need to support withdrawing from the auto one, while it's possible to cover the withdrawing case without storing the mapping (for example, store the denom in the contract), I feel it's safer to keep that mapping for now, maybe some other use case will need it?

Reply:
Actually, the comment on this function is misleading, fixed.

Reply:
ok but lets try to keep the auto deployment logic confined and lets not make too many special case otherwise it might be confusing to maintain